doc/jtag_dmi.md

```diff
@@ -0,0 +1,169 @@
+# JTAG Debug Module Interface
```

Comment:
DMI is the interface, DTM is the transport itself. The code is already rather confused/wrong about this, let's not make it worse :smile: 

```suggestion
# JTAG Debug Transport Module
```


A more general comment: I would (prefer) including this documentation in the debug-system.md file to have it all in one place.